service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/CreateAzureNetworkInterfaceStepTest.java

```diff
@@ -0,0 +1,125 @@
+package bio.terra.workspace.service.resource.controlled.cloud.azure.vm;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.when;
+
+import bio.terra.landingzone.library.landingzones.deployment.SubnetResourcePurpose;
+import bio.terra.workspace.amalgam.landingzone.azure.LandingZoneApiDispatch;
+import bio.terra.workspace.app.configuration.external.AzureConfiguration;
+import bio.terra.workspace.common.BaseUnitTest;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.generated.model.ApiAzureLandingZoneDeployedResource;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.resource.controlled.cloud.azure.network.ControlledAzureNetworkResource;
+import bio.terra.workspace.service.resource.model.WsmResourceType;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import com.azure.resourcemanager.network.NetworkManager;
+import com.azure.resourcemanager.network.models.Network;
+import com.azure.resourcemanager.network.models.Networks;
+import com.azure.resourcemanager.network.models.Subnet;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.test.context.ActiveProfiles;
+
+@ActiveProfiles("unit")
+public class CreateAzureNetworkInterfaceStepTest extends BaseUnitTest {
+
+  @Mock private NetworkManager networkManager;
+
+  @Mock private AzureConfiguration azureConfiguration;
+  @Mock private CrlService crlService;
+  @Mock private ControlledAzureVmResource resource;
+
+  @Mock private ResourceDao resourceDao;
+
+  @Mock private LandingZoneApiDispatch landingZoneApiDispatch;
+
+  @Mock private ControlledAzureNetworkResource controlledAzureNetworkResource;
+
+  @Mock private Networks networks;
+
+  @Mock private AzureCloudContext azureCloudContext;
+
+  private final String STUB_MRG = "RG";
+
+  private final String STUB_SUBNET = "SUBNET";
+
+  private CreateAzureNetworkInterfaceStep networkInterfaceStep;
+
+  @Mock private Network armNetwork;
+
+  @Mock private Subnet armSubnet;
+
+  @BeforeEach
+  void setUp() {
+    networkInterfaceStep =
+        new CreateAzureNetworkInterfaceStep(
+            azureConfiguration, crlService, resource, resourceDao, landingZoneApiDispatch);
+    when(networkManager.networks()).thenReturn(networks);
+    var subnets = new HashMap<String, Subnet>();
+    subnets.put(STUB_SUBNET, armSubnet);
+    when(armNetwork.subnets()).thenReturn(subnets);
+  }
+
+  @Test
```

Comment:
These two tests look almost identical except setup part. Should we somehow differentiate the asserts? Maybe to have different name/id?
The method under test is not public. Q: 1) Do we have default modifier for this method only for testing purposes? 2) Should we also test getNetworkResourcesFromLandingZone (since it's a part of the previous method)?

Reply:
Mockito uses strict stubbing. If the interaction as outlined in the setup is not used, then the test will fail. The assertion confirms that through the expected calls the method returns the expected network resources. The method is scoped to the package, because it should not be public, and since there was no test for this step, this was the best way I found to test my change without writing an overall test.